CleverPush/Source/CleverPushInstance.m

```diff
@@ -1475,6 +1475,12 @@ - (void)handleNotificationOpened:(NSDictionary*)payload isActive:(BOOL)isActive
         }
     }
     if (notification != nil && [notification objectForKey:@"appBanner"] != nil && ![[notification objectForKey:@"appBanner"] isKindOfClass:[NSNull class]]) {
+        if ([notification objectForKey:@"voucherCode"] != nil && ![[notification objectForKey:@"voucherCode"] isKindOfClass:[NSNull class]] && ![[notification objectForKey:@"voucherCode"] isEqualToString:@""]) {
+            NSMutableDictionary *voucherCode = [[NSMutableDictionary alloc] init];
+            [voucherCode setObject:[notification objectForKey:@"voucherCode"] forKey:[notification objectForKey:@"appBanner"]];
+            [CPAppBannerModuleInstance setCurrentVoucherCodePlaceholder:voucherCode];
```

Comment:
we need new names
voucherCode here represents a dictionary of appBannerIDToVoucherCode, it cant be named vouchercode
also we need to check if there is already a dictionary and if so we need to add a new entry, currently we are just overriding the dictionary every time
mabye @gbhrdt has ideas for some good names

Reply:
1: I have changed the name of the dictionary to `appBannerVoucherCode` from `voucherCode`. That name is good?.

2: I have changed some logic here, like first we will check that any previous record or ID is in the dictionary, then the new record will be added. If there is no previous record, then it will be added as the first record in that dictionary.

3: Every time we launch the application, those previous records will be cleared because we are not storing them in the preference data. Should we also handle the preference data?


[https://ibb.co/xMXQ61V](https://ibb.co/xMXQ61V)

Can you please check the above image, and if we are on the same page, then I will make this change in the PR?

Reply:
voucherCodesByAppBanner